test/fixtures/for-in/assign-enhanced-in/expected.js

```diff
@@ -0,0 +1,3 @@
+for (const k in obj = complexFunction()) {
+  const v = (obj = complexFunction())[k];
```

Comment:
ideally let's detect this is happening and have the output be simply `obj[k]`. 

Would also be fine to compile to something like
```js
for (const k in _ref = obj = complexFunction()) {
  const v = _ref[k];
}
```